libraries/botbuilder-adapter-twilio-whatsapp/src/twilioWhatsAppAdapter.ts

```diff
@@ -329,22 +374,28 @@ export class TwilioWhatsAppAdapter extends BotAdapter {
         // Handle mentions
         // Not supported by current Twilio WhatsApp API yet
 
-        // Handle locations
-        // Not supported by current Twilio WhatsApp API yet
-
         // Create new Message for Twilio
         // @ts-ignore Using any since MessageInstance interface doesn't include `mediaUrl`
         const message: any = {
             body: activity.text,
             from: this.settings.phoneNumber,
-            to: activity.conversation.id,
-            mediaUrl: undefined
+            to: activity.conversation.id
         };
 
+        // Handle Persistant Accations (like locations)
+        // https://www.twilio.com/docs/sms/whatsapp/api#location-messages-with-whatsapp
+        if (activity?.channelData?.persistentAction) {
+            if (Array.isArray(activity.channelData.persistentAction)) {
+                message.persistentAction = activity.channelData.persistentAction;
+            } else {
+                message.persistentAction = [activity.channelData.persistentAction];
+            }
+        }
+
         // Handle attachments
         // One media attachment is supported per message, with a size limit of 5MB.
         // https://www.twilio.com/docs/sms/whatsapp/api#sending-a-freeform-whatsapp-message-with-media-attachment
-        if (activity.attachments && activity.attachments.length > 0) {
+        if (activity?.attachments?.length > 0) {
```

Comment:
Does this work now? Do we need to update the `engines` property in the `package.json` to reflect the version that supports this?

Reply:
I did bump typescript to `^3.7.4`, which supports optional chaining. Since it is transpiled by Typescript, it doesn't matter which node version you run.